### PR TITLE
Member resolving fix for derived attributes (#62937)

### LINKED ISSUE #62937: System.Text.Json source generator fails if [Required(ErrorMessage = "...")] attribute is present.
### Description

I tried to use the Json source generator with a class which has a [Required] attribute with an error message, but the generator succeeds only when the ErrorMessage is not specified. I tried it with other Validation attributes (StringLength, Range), and I got the same result.

### Reproduction Steps

Add the following two classes in a .NET 6.0 project, then try to build it.

```csharp
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

public class TestDto
{
    [Required(ErrorMessage = "asdf")]
    public string Name { get; set; }
}

[JsonSerializable(typeof(TestDto))]
public partial class Serializers : JsonSerializerContext
{

}

```

### Expected behavior

The source generator should generate the properties and the build should succeed.

### Actual behavior

The source generator and the build fails and I get an error in the Output window:

```
Build started...
1>------ Build started: Project: SystemTextJsonTest, Configuration: Debug Any CPU ------
1>CSC : warning CS8785: Generator 'JsonSourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'InvalidOperationException' with message 'Sequence contains no elements'
1>D:\src\SystemTextJsonTest\SystemTextJsonTest\TestDto.cs(11,22,11,33): error CS0534: 'Serializers' does not implement inherited abstract member 'JsonSerializerContext.GetTypeInfo(Type)'
1>D:\src\SystemTextJsonTest\SystemTextJsonTest\TestDto.cs(11,22,11,33): error CS0534: 'Serializers' does not implement inherited abstract member 'JsonSerializerContext.GeneratedSerializerOptions.get'
1>Done building project "SystemTextJsonTest.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

### Regression?

_No response_

### Known Workarounds

Remove the ErrorMessage, ErrorMessageResourceName and ErrorMessageResourceType properties from the validation attributes.

### Configuration

.NET 6.0.100
Microsoft Windows [Version 10.0.19043.1348]
x64

### Other information

_No response_

### PR DESCRIPTION
**Pull request by Kloizdena**
Fixes #62937

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs [hunk 1]
```diff
@@ -158,4 +158,23 @@ public struct PersonStruct
         public string FirstName { get; set; }
         public string LastName { get; set; }
     }
+
+    public class TypeWithValidationAttributes
+    {
+        [ComponentModel.DataAnnotations.Required(ErrorMessage = "Name is required")]
```
Comment (layomia) [inline src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs:164]: Can you add another property where with a `[Required]` attribute but no properties?

Comment (Kloizdena) [reply src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs:164]: OK, done.

File: src/libraries/System.Text.Json/gen/Reflection/CustomAttributeDataWrapper.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public CustomAttributeDataWrapper(AttributeData a, MetadataLoadContextInternal m
             var namedArguments = new List<CustomAttributeNamedArgument>();
             foreach (KeyValuePair<string, TypedConstant> na in a.NamedArguments)
             {
-                var member = a.AttributeClass!.GetMembers(na.Key).First();
+                var member = a.AttributeClass.BaseTypes().SelectMany(t => t.GetMembers(na.Key)).First();
```
Comment (layomia) [inline src/libraries/System.Text.Json/gen/Reflection/CustomAttributeDataWrapper.cs:18]: Ideally `a.AttributeClass.BaseTypes()` should be saved in a local above, but that can be included in a follow-up.

**Review by layomia — APPROVED** [review]
LGTM, thanks!

**Pull request merged**